src/LogMonitorEntry.js

```diff
@@ -31,6 +31,7 @@ export default class LogMonitorEntry extends Component {
   constructor(props) {
     super(props);
     this.handleActionClick = this.handleActionClick.bind(this);
+    this.previousState = this.props.previousState || this.props.state;
```

Comment:
In general, this seems to give you the wrong semantics. You are making it look like the previous state *is* the current state for the first action. However what really happens is that it doesn’t exist.